rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala

```diff
@@ -791,24 +791,184 @@ trait HotStoreSpec[F[_], M[_]] extends FlatSpec with Matchers with GeneratorDriv
         .map(_._1)
     }
   }
+
+  /*
+   *  The tests below were migrated from IStoreTests to check backward compatibility
+   */
+
+  "putDatum" should "put datum in a new channel" in forAll("channel", "datum") {
+    (channel: String, datumValue: String) =>
+      fixture { (_, _, hotStore) =>
+        val key   = channel
+        val datum = Datum.create(channel, datumValue, false)
+
+        for {
+          _   <- hotStore.putDatum(key, datum)
+          res <- hotStore.getData(key)
+          _   <- S.delay { res should contain theSameElementsAs (Seq(datum)) }
+        } yield ()
+      }
+  }
+
+  it should "append datum if channel already exists" in forAll("channel", "datum") {
```

Comment:
this test and the previous one would be better suited to a generative test that puts random datum items in a channel

Reply:
Thanks for the review @odeac! There already exist generative tests for `HotStore`. The purpose of this PR is adaptation of the *existing* `IStoreTests` (without changing them too much) and running them against `HotStore` in order to check whether the new impl preserves the behaviour